packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -37,22 +38,27 @@ export default {
     ],
   },
   create(context) {
+    const inputOptions = context.options && context.options[0];
     // Parse the `additionalHooks` regex.
     const additionalHooks =
-      context.options &&
-      context.options[0] &&
-      context.options[0].additionalHooks
-        ? new RegExp(context.options[0].additionalHooks)
+      inputOptions && inputOptions.additionalHooks
+        ? new RegExp(inputOptions.additionalHooks)
+        : undefined;
+
+    // Parse the `safeHooks` regex
+    const safeHooks =
+      inputOptions && inputOptions.safeHooks
+        ? new RegExp(inputOptions.safeHooks)
         : undefined;
 
     const enableDangerousAutofixThisMayCauseInfiniteLoops =
-      (context.options &&
-        context.options[0] &&
-        context.options[0].enableDangerousAutofixThisMayCauseInfiniteLoops) ||
+      (inputOptions &&
+        inputOptions.enableDangerousAutofixThisMayCauseInfiniteLoops) ||
```

Comment:
We were accessing `context.options && context.options[0]` a bunch here so I pulled that into a var. Other than that I just added logic to extract `safeHooks`.